static/js/apps/eval_retrieval_generation/query_section.tsx

```diff
@@ -104,6 +105,58 @@ function getAnswer(
     });
 }
 
+/**
+ * Absolutely positions the tooltip within the given annotation so it fits
+ * within the parent query section.
+ */
+function adjustTooltipPosition(annotationEl: Element): void {
+  const tooltipEl = annotationEl.querySelector(
+    ".dc-stat-tooltip"
+  ) as HTMLDivElement;
+  const parentSection = annotationEl.closest("#query-section");
+  const sectionRect = parentSection.getBoundingClientRect();
+
+  // Limit tooltip width to section width.
+  if (tooltipEl.getBoundingClientRect().width > sectionRect.width - 2) {
```

Comment:
nit: curious what the 2 is, is that just a margin? if so, maybe can make that a constant

Reply:
Updating to 2 * sectionBorderWidth.